tests/src/test/java/com/maxio/advancedbilling/controllers/components/create/ComponentsControllerCreateEventsBasedTest.java

```diff
@@ -0,0 +1,236 @@
+                .filter(pp -> pp.getName().equals(expectedPricePoint.getName()))
+                .findFirst().get();
+        assertThat(pricePoint.getId()).isNotNull();
+        assertThat(pricePoint.getComponentId()).isEqualTo(component.getId());
+        assertThat(pricePoint.getName()).isEqualTo(expectedPricePoint.getName());
+        assertThat(pricePoint.getHandle()).isEqualTo(expectedPricePoint.getHandle());
+        assertThat(pricePoint.getType()).isEqualTo(PricePointType.CATALOG);
+        assertThat(pricePoint.getPricingScheme()).isEqualTo(PricingScheme.VOLUME);
+
+        assertThat(pricePoint.getPrices()).hasSize(2);
+        ComponentPrice componentPrice1 = pricePoint.getPrices().get(0);
+        assertThat(componentPrice1.getId()).isNotNull();
+        assertThat(componentPrice1.getComponentId()).isEqualTo(component.getId());
+        assertThat(componentPrice1.getStartingQuantity()).isEqualTo(0);
+        assertThat(componentPrice1.getEndingQuantity()).isEqualTo(5);
+        assertThat(componentPrice1.getUnitPrice()).isEqualTo(String.valueOf(catalogPricePointPrice1));
+        assertThat(componentPrice1.getPricePointId()).isEqualTo(pricePoint.getId());
+        assertThat(componentPrice1.getFormattedUnitPrice()).isEqualTo("$103.00");
+        assertThat(componentPrice1.getSegmentId()).isNull();
+        ComponentPrice componentPrice2 = pricePoint.getPrices().get(1);
+        assertThat(componentPrice2.getId()).isNotNull();
+        assertThat(componentPrice2.getComponentId()).isEqualTo(component.getId());
+        assertThat(componentPrice2.getStartingQuantity()).isEqualTo(6);
+        assertThat(componentPrice2.getEndingQuantity()).isNull();
+        assertThat(componentPrice2.getUnitPrice()).isEqualTo(String.valueOf(catalogPricePointPrice2));
+        assertThat(componentPrice2.getPricePointId()).isEqualTo(pricePoint.getId());
+        assertThat(componentPrice2.getFormattedUnitPrice()).isEqualTo("$1.523");
+        assertThat(componentPrice2.getSegmentId()).isNull();
+
+        assertThat(pricePoint.getCreatedAt()).isNotNull();
+        assertThat(pricePoint.getUpdatedAt()).isNotNull();
+        assertThat(pricePoint.getArchivedAt()).isNull();
+        assertThat(pricePoint.getUseSiteExchangeRate()).isTrue();
+        assertThat(pricePoint.getTaxIncluded()).isFalse();
+        assertThat(pricePoint.getSubscriptionId()).isNull();
+        assertThat(pricePoint.getInterval()).isNull();
+        assertThat(pricePoint.getIntervalUnit()).isNull();
+        assertThat(pricePoint.getCurrencyPrices()).isNull();
+    }
+
+    @Test
+    void shouldReturn422WhenInputDataIsIncorrect() {
+        // given
+        EBBComponent ebbComponent = new EBBComponent.Builder()
+                .name("Events Based Component " + randomNumeric(10))
+                .handle("wrong handle")
+                .pricingScheme(PricingScheme.VOLUME)
+                .build();
+
+        // when - then
+        assertUnprocessableEntity(
```

Comment:
You can use assertThatErrorListResponse and hasErrors.

Reply:
done